src/ConnectedMode/Persistence/ISolutionBindingRepository.cs

```diff
@@ -70,5 +86,42 @@ public BoundSonarQubeProject Read(string configFilePath)
 
             return bound;
         }
+
+        public bool Write(string configFilePath, BoundSonarQubeProject binding)
+        {
+            _ = binding ?? throw new ArgumentNullException(nameof(binding));
+
+            if (string.IsNullOrEmpty(configFilePath))
+            {
+                return false;
+            }
+
+            if (!solutionBindingFileLoader.Save(configFilePath, binding))
+            {
+                return false;
+            }
+
+            credentialsLoader.Save(binding.Credentials, binding.ServerUri);
+
+            return true;
+        }
+
+        public IEnumerable<BoundSonarQubeProject> List()
+        {
+            var result = new List<BoundSonarQubeProject>();
```

Comment:
there's no need for list if you return IEnumerable. Also, for now we don't need to list the full list of bindings with QPs, we only need the connections. So, to simplify things, I would directly parse the connection info and return just that, unless you see a future need to return the full info.

Reply:
I don't see a future use but to keep things consistent using the same object with the rest of the methods makes more sense for me.

Reply:
I don't think it's worth it to parse unnecessary data just to keep the interface slightly more clean. There are ways to minimize the difference between models and the methods used to parse them (generics, inheritance hierarchy of classes, etc.)

Reply:
Using something like Generics... etc, would be an overkill. 99% of the time, this collection will have only one member and  even if there are multiple connections, it won't be hundreds or even tens of connections.

Reply:
I don't see generics as something that is too complicated to implement. And anyways, I was only using it as an example.